### PR TITLE
let `start` open anything and everything

### LINKED ISSUE #7546: Add open . command works for opening directory
### Related problem

Currently, NuShell does not support the ability to open the current directory with the 'open .' command. 

### Describe the solution you'd like

This is a common feature in other shells and can be useful for quickly viewing and manipulating the contents of the current directory. I propose adding this functionality to NuShell by modifying the 'open' command to accept a '.' argument, which would open the current directory in the default file manager.

### Describe alternatives you've considered

_No response_

### Additional context and details

_No response_

### PR DESCRIPTION
**Pull request by merelymyself**
# Description

Fixes #7546 's request. I'm unsure, so hopefully someone in charge of design can chip in.

# User-Facing Changes

`open` now opens directories in the default file manager.

# Tests + Formatting

Don't forget to add tests that cover your changes.

Make sure you've run and fixed any issues with these commands:

- `cargo fmt --all -- --check` to check standard code formatting (`cargo fmt --all` applies these changes)
- `cargo clippy --workspace -- -D warnings -D clippy::unwrap_used -A clippy::needless_collect` to check that you're using the standard code style
- `cargo test --workspace` to check that all tests pass

# After Submitting

If your PR had any user-facing changes, update [the documentation](https://github.com/nushell/nushell.github.io) after the PR is merged, if necessary. This will help us keep the docs up to date.

### CODE REVIEW

**Comment by rgwood** [comment]
Thanks for the PR. I don't think we should add this functionality to `open` - I don't like the idea of giving 1 command 2 very different jobs.

IIRC Nu used to have this functionality in a `start` plugin but that didn't get ported for the v0.60 rewrite. It used this crate: https://github.com/Byron/open-rs

We might be open to bringing this functionality back but I don't think it should happen in `open`.

**Comment by fdncred** [comment]
ya, I'd support a `start` plugin like we had before.

**Comment by rgwood** [comment]
@fdncred do you prefer a plugin to a built-in command?

**Comment by fdncred** [comment]
@rgwood my tendency is to go with a plugin because that's what we had before. someone made that decision, i don't remember who. but it wouldn't hurt my feelings to have it built-in either.

**Comment by rgwood** [comment]
I think I’d prefer a built-in command; it should be very low on

**Comment by rgwood** [comment]
Sorry, pressed the wrong button on my phone and closed this a bit too early.

**Comment by fdncred** [comment]
@rgwood I would want to use the original crate though. It does quite a bit more checking and looks like it makes pretty good decisions per platform.

File: crates/nu-command/src/filesystem/open_dir.rs [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+        if path.is_dir() {
+            open::that(path)?;
+            Ok(PipelineData::Empty)
+        } else {
+            Err(ShellError::DirectoryNotFound(call_span, None))
+        }
+    }
+
+    fn examples(&self) -> Vec<nu_protocol::Example> {
```
Comment (fdncred) [inline crates/nu-command/src/filesystem/open_dir.rs:84]: These examples look like they're for open vs open-dir

**Comment by fdncred** [comment]
Not sure how I feel about `open-dir` vs `start`. I could probably live with it if others can since it says exactly what it's supposed to do where start is kind of ambiguous.

**Comment by rgwood** [comment]
I don’t like the name open-dir because the command should be able to open files as well as directories.

**Comment by fdncred** [comment]
> I don’t like the name open-dir

Does that mean your proposal is to rename it to `start` or did you have something else in mind?

**Comment by rgwood** [comment]
I think `start` is a good name but I'd be open to other suggestions.

**Review by fdncred — APPROVED** [review]
LGTM! Thanks

**Review by rgwood — APPROVED** [review]
Looks good. Thanks!

**Pull request merged**